src/index.js

```diff
@@ -48,6 +50,7 @@ class Dropzone extends React.Component {
       document.addEventListener('dragover', Dropzone.onDocumentDragOver, false)
       document.addEventListener('drop', this.onDocumentDrop, false)
     }
+    this.fileInputEl.addEventListener('click', this.onInputElementClick, false)
```

Comment:
I tried this 

```
    return (
      <div
        className={className}
        style={appliedStyle}
        {...divProps /* expand user provided props first so event handlers are never overridden */}
        onClick={this.onClick}
        onDragStart={this.onDragStart}
        onDragEnter={this.onDragEnter}
        onDragOver={this.onDragOver}
        onDragLeave={this.onDragLeave}
        onDrop={this.onDrop}
        ref={this.setRef}
      >
        {this.renderChildren(children, isDragActive, isDragReject)}
        <input
          {...{
            ...inputProps,
            onClick: evt => {
              evt.stopPropagation()
              if (typeof inputProps.onClick === 'function') {
                inputProps.onClick()
              }
            }
          }}
          {...inputAttributes}
        />
      </div>
    )
```

However I still got the double click, and the test I created for this also failed.